msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerContentProviderStrategy.java

```diff
@@ -0,0 +1,487 @@
+                                        @Nullable final String negotiatedBrokerProtocolVersion)
+            throws BaseException {
+        final String methodName = "getBrokerAuthorizationIntentForContentProvider";
+
+        return performContentProviderOperation(parameters, new ContentProviderOperation<InteractiveTokenCommandParameters, Intent>() {
+
+            @Override
+            public String getRequestString(InteractiveTokenCommandParameters parameters) {
+                return null; // broker returns us an intent based on calling uid , no request string needed
+            }
+
+            @Override
+            public String[] getOptions() {
+                return new String[]{negotiatedBrokerProtocolVersion};
+            }
+
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @Override
+            public String getUriPath() {
+                return ACQUIRE_TOKEN_INTERACTIVE_PATH;
+            }
+
+            @Override
+            public Intent getResultFromBundle(Bundle resultBundle) throws BaseException {
+                final Bundle requestBundle = mRequestAdapter.getRequestBundleForAcquireTokenInteractive(
+                        parameters,
+                        negotiatedBrokerProtocolVersion
+                );
+                Intent interactiveRequestIntent = new Intent();
+                interactiveRequestIntent.putExtras(requestBundle);
+                interactiveRequestIntent.putExtras(resultBundle);
+                interactiveRequestIntent.setPackage(resultBundle.getString(BROKER_PACKAGE_NAME));
+                interactiveRequestIntent.setClassName(
+                        resultBundle.getString(BROKER_PACKAGE_NAME, ""),
+                        resultBundle.getString(BROKER_ACTIVITY_NAME, "")
+                );
+                interactiveRequestIntent.putExtra(
+                        AuthenticationConstants.Broker.NEGOTIATED_BP_VERSION_KEY,
+                        negotiatedBrokerProtocolVersion
+                );
+                return interactiveRequestIntent;
+            }
+        });
+    }
+
+    @Override
+    AcquireTokenResult acquireTokenSilent(@NonNull final SilentTokenCommandParameters parameters,
+                                          @Nullable final String negotiatedBrokerProtocolVersion)
+            throws BaseException {
+        final String methodName = "acquireTokenSilentWithContentProvider";
+
+        return performContentProviderOperation(
+                parameters,
+                new ContentProviderOperation<SilentTokenCommandParameters, AcquireTokenResult>() {
+                    @Nullable
+                    @Override
+                    public String getRequestString(SilentTokenCommandParameters parameters) {
+                        final BrokerRequest brokerRequest = mRequestAdapter.
```

Comment:
since we have specific broker request classes now for other operations such as hello, remove account etc. I would suggest renaming `BrokerRequest.java` to something like `BrokerSilentTokenRequest.java` as I believe this will now be only used for silent token requests?

Reply:
`BrokerRequest` will be used to pass all the request parameters for both silent and interactive requests. Given we need to support back compat with old MSAL and old brokers. I am not inclined  to split and create two new classes for `silent` and `interactive` , so leaving as - is